### PR TITLE
fix: after we init the config file we need to read it in

### PR DESCRIPTION
**Pull request by frrist**
- this will ensure the written config file's value are persisted to vipers defaults

Example of this fixing the issue:
```
frrist@oak ~/W/s/g/b/bacalhau (frrist/fix-config-init)> rm -r ~/.bacalhau/

frrist@oak ~/W/s/g/b/bacalhau (frrist/fix-config-init)> env BACALHAU_ENVIRONMENT=staging ./bin/darwin_arm64/bacalhau version
11:42:19.832 | INF pkg/setup/setup.go:36 > using $HOME for bacalhau repo [repo:/Users/frrist/.bacalhau]
11:42:19.832 | INF pkg/repo/fs.go:187 > Initializing repo at '/Users/frrist/.bacalhau' for environment 'staging'
 CLIENT               SERVER 
 v1.0.4-12-g08e65db8  v1.0.4 
 
frrist@oak ~/W/s/g/b/bacalhau (frrist/fix-config-init)> ./bin/darwin_arm64/bacalhau version
11:42:46.146 | INF pkg/setup/setup.go:36 > using $HOME for bacalhau repo [repo:/Users/frrist/.bacalhau]
 CLIENT               SERVER 
 v1.0.4-12-g08e65db8  v1.0.4 
```

### CODE REVIEW

**Review by rossjones — APPROVED** [review]

File: pkg/config/types/generated_constants.go [hunk 1]
```diff
@@ -72,6 +72,8 @@ const NodeComputeQueueExecutorBufferBackoffDuration = "Node.Compute.Queue.Execut
 const NodeComputeLogging = "Node.Compute.Logging"
 const NodeComputeLoggingLogRunningExecutionsInterval = "Node.Compute.Logging.LogRunningExecutionsInterval"
 const NodeRequester = "Node.Requester"
+const NodeRequesterJobDefaults = "Node.Requester.JobDefaults"
```
Comment (frrist) [inline pkg/config/types/generated_constants.go:75]: As the commit message for this change calls out. These values were changed in https://github.com/bacalhau-project/bacalhau/pull/2791 but we forgot to regenerate the config constants for it. Addressing here.

Comment (frrist) [reply pkg/config/types/generated_constants.go:75]: We should probably add a CI check for this..

Comment (frrist) [reply pkg/config/types/generated_constants.go:75]: Added here: https://github.com/bacalhau-project/bacalhau/pull/2828

**Comment by frrist** [comment]
I found another bug in our config code, could use a second round of review. Thankfully it was also a fairly simple fix.

**Pull request merged**